tests/coords/transform_coords_test.py

```diff
@@ -1003,3 +1004,35 @@ def test_raises_when_keyword_syntax_clashes_with_options(option):
     da = sc.data.table_xyz(nrow=10)
     with pytest.raises(TypeError):
         da.transform_coords(**{option: lambda x: x})
+
+
+def test_works_with_partial():
+
+    def f(a, x):
+        return a * x
+
+    g = functools.partial(f, sc.scalar(5))
+    da = sc.data.table_xyz(nrow=10)
+    assert 'ax' in da.transform_coords(ax=g).coords
+
+
+def test_works_with_class_defining___call__():
+
+    class A:
+
+        def __call__(self, x):
+            return x * x
+
+    da = sc.data.table_xyz(nrow=10)
+    assert 'xx' in da.transform_coords(xx=A()).coords
+
+
+def test_input_coords_can_be_defined_via_property():
```

Comment:
But how would this work with `Lookup`?

Reply:
I guess, `Lookup` would inherit from `Node`. Not very nice...
I was somehow under the impression that `Lookup` would be able to do something on its end. But it doesn't know that it is being passed to `transform_coords`.

What about a more unique name like `__transform_coords_inputs__` to avoid potential (unlikely) conflicts?

Reply:
Yes makes sense. You it be a property or a function returning a list?

Reply:
See update, I kept it as a property.

Reply:
Looks good